### DISCUSSION TITLE
Having trouble displaying command options in help...

### DISCUSSION THREAD

**Discussion by jbizios** [spectreconsole/spectre.console#1208] [Q&A] [answered]
I have written a utility with Command Option specified  as:

 public sealed class ConsoleCommand : AsyncCommand<ConsoleSettings>
    {
        public override Task<int> ExecuteAsync(CommandContext context, ConsoleSettings settings)
        {
             // Some work
             return Task.FromResult(1);
       }
   }
And Console Settings as:

 public sealed class ConsoleSettings : CommandSettings
    {
        [CommandOption("-s|--seconds", IsHidden = false)]
        **[Description("How often to refresh the values in seconds")]**
        public int? Seconds { get; set; }

        public override ValidationResult Validate()
        {
            if (!Seconds.HasValue) return ValidationResult.Success();

            return Seconds.Value > 0
                ? ValidationResult.Success()
                : ValidationResult.Error("Seconds must be greater than 0");
        }
    }

I add the command in Main as:
          var app = new CommandApp();

            app.Configure(config =>
            {
                 config.AddCommand<ConsoleCommand>("console")
                    .WithDescription("Sends all Data to the Console Window")
                    .WithExample(new[] { "console", "--seconds", "5" });
            }

When I render help I get:

USAGE:
    MyUtility.exe [OPTIONS] <COMMAND>

EXAMPLES:
    MyUtility.exe console --seconds 5   

OPTIONS:
    -h, --help       Prints help information
    -v, --version    Prints version information

COMMANDS:
    console    Sends all Data to the Console Window
    
I expected to see the Command Setting of "**_seconds_**" as an option with the command "**_console_**" in the help screen and the description of "**_How often to refresh the values in seconds_**" on that option.

Is this not supported or am I doing something wrong?

Thanks in advance for any help or guidance!

**Comment by patriksvensson** [comment, marked as answer]
You will need to return the help for the command, such as `myapp.exe console --help` or promote ConsoleCommand as the default command.

**Reply by jbizios** [reply]
Great.  That worked, I didn't see that in the documentation.

How do you promote the command as default?

Thanks again!

**Reply by patriksvensson** [reply]
You ser the default command by specifying it when creating the `CommandApp` instance.

```csharp
var app = new CommandApp<ConsoleCommand>();
```

**Reply by jbizios** [reply]
Found it.  Thanks again!

Congrats on this library!
